src/uwtools/api/config.py

```diff
@@ -0,0 +1,77 @@
+# pylint: disable=unused-import
+
+from typing import Optional
+
+from uwtools.config.atparse_to_jinja2 import convert as _convert_atparse_to_jinja2
+from uwtools.config.tools import compare_configs as _compare
+from uwtools.config.tools import realize_config as _realize
+from uwtools.config.validator import validate_yaml as _validate
+from uwtools.types import DefinitePath, OptionalPath
+from uwtools.utils.file import FORMAT
+
+
+def compare(
+    config_a_path: DefinitePath,
+    config_a_format: str,
+    config_b_path: DefinitePath,
+    config_b_format: str,
+) -> bool:
+    """
+    ???
+    """
+    return _compare(
+        config_a_path=config_a_path,
+        config_a_format=config_a_format,
+        config_b_path=config_b_path,
+        config_b_format=config_b_format,
+    )
+
+
+def realize(
+    input_file: OptionalPath,
```

Comment:
I had previously thought that the API should insist on files and not support streaming IO, but two things changed my mind:
1. Since the CLI has been updated to use the API functions as its interface to the rest of the `uwtools` library, and since the CLI needs to support both file and streaming IO, the API needs to support streams, too. Otherwise, the CLI would need to duplicate much of what's here, only with type changes for the inputs and outputs that should support streaming. That seems like it'd be an unfortunate DRY violation.
2. A user might well like to use streaming IO, too -- for example to create their own CLI based on the `uwtools` API.

So, everything in this PR continues to support the ability to use ether file or streaming IO.